kuma/core/i18n.py

```diff
@@ -1,4 +1,165 @@
+    as the provided language codes are taken from the HTTP request. See also
+    <https://www.djangoproject.com/weblog/2007/oct/26/security-fix/>.
+
+    Based on Django 1.8.19's get_supported_language_variant from
+    django/utils/translation/trans_real.py, with changes:
+
+    * Language code can also be a Kuma language code
+    * Return Kuma languge codes
+    * Force strict=False to always allow fuzzy matching (zh-CHS gets zh-CN)
+    """
+    if raw_lang_code:
+        lang_code = kuma_language_code_to_django(raw_lang_code)
+
+        # Check for known override
+        if lang_code in settings.LOCALE_ALIASES:
+            return settings.LOCALE_ALIASES[lang_code]
+
+        # If 'fr-ca' is not supported, try special fallback or language-only 'fr'.
+        possible_lang_codes = [lang_code]
+        try:
+            possible_lang_codes.extend(LANG_INFO[lang_code]['fallback'])
+        except KeyError:
+            pass
+        generic_lang_code = lang_code.split('-')[0]
+        possible_lang_codes.append(generic_lang_code)
+        raw_supported_lang_codes = get_languages()
+        supported_lang_codes = [kuma_language_code_to_django(lang)
+                                for lang in raw_supported_lang_codes]
+
+        # Look for exact match
+        for code in possible_lang_codes:
+            if code in supported_lang_codes and check_for_language(code):
+                return django_language_code_to_kuma(code)
+        # If fr-fr is not supported, try fr-ca.
+        for supported_code in supported_lang_codes:
+            if supported_code.startswith(generic_lang_code + '-'):
+                return django_language_code_to_kuma(supported_code)
+    raise LookupError(raw_lang_code)
+
+
+def get_language_from_path(path):
+    """
+    Returns the language-code if there is a valid language-code
+    found in the `path`.
+
+    If `strict` is False (the default), the function will look for an alternative
+    country-specific variant when the currently checked is not found.
+
+    Based on Django 1.8.19's get_language_from_path from
+    django/utils/translation/trans_real.py, with changes:
+
+    * Don't accept or pass strict parameter (assume strict=False).
+    """
+    regex_match = language_code_prefix_re.match(path)
+    if not regex_match:
+        return None
+    lang_code = regex_match.group(1)
+    try:
+        return get_supported_language_variant(lang_code)
+    except LookupError:
+        return None
+
+
+def get_language_from_request(request):
+    """
+    Analyzes the request to find what language the user wants the system to
+    show. Only languages listed in settings.LANGUAGES are taken into account.
+    If the user requests a sublanguage where we have a main language, we send
+    out the main language.
+
+    If check_path is True, the URL path prefix will be checked for a language
+    code, otherwise this is skipped for backwards compatibility.
```

Comment:
Super nit. This `check_path` portion seems out of place now that it has been removed from the function signature.